test/unit/source/deinitialization.d

```diff
@@ -0,0 +1,150 @@
+// global.deinitialize
+unittest
+{
+    import dmd.globals : global;
+
+    static void assertStructsEqual(T)(const ref T a, const ref T b) @nogc pure nothrow
+        if (is(T == struct))
+    {
+        foreach (i, _; typeof(a.tupleof))
+        {
+            enum name = __traits(identifier, a.tupleof[i]);
+
+            static if (is(typeof(a.tupleof[i]) == const(char*)))
+                assert(a.tupleof[i].fromStringz == b.tupleof[i].fromStringz, name);
+            else
+                assert(a.tupleof[i] == b.tupleof[i], name);
+        }
+    }
+
+    immutable init = global.init;
+    assertStructsEqual(global, init);
+
+    global._init();
```

Comment:
Partially OT: What do you think about renaming this to `initialize`, s.t. the pattern matches with the other modules?

Reply:
And also for the same reason we renamed `TypeInfo.init` to `TypeInfo.initializer`.

Reply:
Yes, I would like to rename all `_init` to `initialize`. That's why I named it `deinitialize` and not `deinit`. This will get rid of the ugly underscore prefix and spell out the full name instead of an abbreviation. A double win, in my opinion 😃.